networkit/cpp/io/NetworkitBinaryWriter.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * NetworkitBinaryWriter.cpp
+ *
+ *@author Charmaine Ndolo <charmaine.ndolo@b-tu.de>
+ */
+
+#include <networkit/io/NetworkitBinaryWriter.hpp>
+
+namespace NetworKit {
+
+NetworkitBinaryWriter::NetworkitBinaryWriter(uint64_t chunks): chunks(chunks) {}
+
+uint64_t NetworkitBinaryWriter::splitNodes(uint64_t nodes, uint64_t chunks) {
+
+	nodesPerChunk = nodes / chunks;
+	return nodesPerChunk;
+}
+
+size_t NetworkitBinaryWriter::encode(uint64_t value, uint8_t* buffer) {
+	uint64_t n;
+	if(!value) {
+		buffer[0] =1;
+		return 1;
+	}
+
+	if (value > (uint64_t(1) << 63)) {
+		n = 8;
+		buffer[0] = 0;
+	} else {
+		n = (64 - __builtin_clzl(value)) / 7;
```

Comment:
`__builtin_clz` is GCC/Clang-only and breaks on Windows. We should use the clz function from TLX (which is already a depending of NetworKit). See https://github.com/tlx/tlx/blob/master/tlx/math/clz.hpp.